lib/src/geometry/circle.dart

```diff
@@ -0,0 +1,95 @@
+import 'dart:math';
+import 'dart:ui';
+
+import '../../components.dart';
+import '../extensions/vector2.dart';
+import 'shape.dart';
+
+class Circle extends Shape {
+  /// The [definition] is how many percentages of [size] that the circle should
+  /// cover.
+  double definition;
+
+  Circle({
+    this.definition = 1.0,
+    Vector2 position,
+    Vector2 size,
+    double angle,
```

Comment:
this seems like a weird to represent a circle. is this really an elipse? what if size.x!=size.y? and angle only matters for non-circle ellipses right. do we need a parameter that just multiplies everything?

I feel we should represent the circle on its fields in it's most pristine form and we can of course have an additional constructor that takes unorthodox parameters and compute stuff and call the main constructor. not the other way around.

Reply:
You're right, this definition is very specific to the collision detection system.
The size is the bounding box that it is defined by, if size is a rectangle it will still be a circle within it, positioned with `position` from the center of of the `size` rectangle.

Reply:
Just changed around the constructors, much better this way.
Will do it for polygon and rectangle too.